File: go-sdk/pkg/models/operations/getfilesidmetadataglobalboxskillscards.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDMetadataGlobalBoxSkillsCardsPathParams struct {
	FileID string `pathParam:"style=simple,explode=false,name=file_id"`
}

type GetFilesIDMetadataGlobalBoxSkillsCardsRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetFilesIDMetadataGlobalBoxSkillsCardsPathParams
}

type GetFilesIDMetadataGlobalBoxSkillsCardsResponse struct {
	ClientError        *shared.ClientError
	ContentType        string
	SkillCardsMetadata *shared.SkillCardsMetadata
	StatusCode         int64
}

File: sdks/ts/src/sdk/models/operations/getfilesidmetadataglobalboxskillscards.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFilesIdMetadataGlobalBoxSkillsCardsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class GetFilesIdMetadataGlobalBoxSkillsCardsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdMetadataGlobalBoxSkillsCardsPathParams;
}


export class GetFilesIdMetadataGlobalBoxSkillsCardsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  skillCardsMetadata?: shared.SkillCardsMetadata;

  @SpeakeasyMetadata()
  statusCode: number;
}
